modules/fluid_properties/include/fluidproperties/IdealGasMixtureFluidProperties.h

```diff
@@ -0,0 +1,200 @@
+  virtual Real want##_from_##prop1##_##prop2(Real prop1, Real prop2, const std::vector<Real> & x)  \
+      const override;                                                                              \
+  virtual ADReal want##_from_##prop1##_##prop2(                                                    \
+      const ADReal & prop1, const ADReal & prop2, const std::vector<ADReal> & x) const override
+
+/**
+ * Class for fluid properties of an ideal gas mixture
+ */
+class IdealGasMixtureFluidProperties : public VaporMixtureFluidProperties, public NaNInterface
+{
+public:
+  static InputParameters validParams();
+
+  IdealGasMixtureFluidProperties(const InputParameters & parameters);
+
+  usingVaporMixtureFluidPropertiesMembers;
+
+  virtual unsigned int numberOfComponents() const override { return _n_components; }
+
+  virtual const SinglePhaseFluidProperties & getPrimaryFluidProperties() const override;
+  virtual const SinglePhaseFluidProperties &
+  getSecondaryFluidProperties(unsigned int i = 0) const override;
+
+  /**
+   * Computes all mass fractions
+   *
+   * @param[in] x_secondary   Secondary mass fractions
+   */
+  template <typename CppType>
+  std::vector<CppType>
+  secondaryToAllMassFractions_templ(const std::vector<CppType> & x_secondary) const;
+  std::vector<ADReal> secondaryToAllMassFractions(const std::vector<ADReal> & x_secondary) const;
+  std::vector<Real> secondaryToAllMassFractions(const std::vector<Real> & x_secondary) const;
+
+  /**
+   * Computes the mixture molar mass
+   *
+   * @param[in] x   All mass fractions
+   */
+  template <typename CppType>
+  CppType mixtureMolarMass_templ(const std::vector<CppType> & x) const;
+  ADReal mixtureMolarMass(const std::vector<ADReal> & x) const;
+  Real mixtureMolarMass(const std::vector<Real> & x) const;
+
+  /**
+   * Computes molar fractions
+   *
+   * @param[in] x   All mass fractions
+   */
+  template <typename CppType>
+  std::vector<CppType> molarFractionsFromMassFractions_templ(const std::vector<CppType> & x) const;
+  std::vector<ADReal> molarFractionsFromMassFractions(const std::vector<ADReal> & x) const;
+  std::vector<Real> molarFractionsFromMassFractions(const std::vector<Real> & x) const;
+
+  override_property(p, v, e);
+  override_property(T, v, e);
+  override_property(v, p, T);
+  override_property(e, p, T);
+  override_property(s, p, T);
+  override_property(c, p, T);
+  override_property(cp, p, T);
+  override_property(cv, p, T);
+  override_property(mu, p, T);
+  override_property(k, p, T);
+
+#undef override_property
+
+protected:
+  /// Names of component fluid properties
+  const std::vector<UserObjectName> _component_fp_names;
+  /// Number of components
+  const unsigned int _n_components;
+  /// Number of secondary components
+  const unsigned int _n_secondary_components;
+
+  /// Component fluid properties objects
+  std::vector<const IdealGasFluidProperties *> _component_fps;
+};
+
+#pragma GCC diagnostic pop
+
+template <typename CppType>
```

Comment:
I dont think these templates need to be in the header since they are used only in that source

Reply:
Oh yeah, good tip, I love me some lean headers!